gammapy/datasets/simulate.py

```diff
@@ -247,13 +248,16 @@ def _sample_coord_time(self, npred, temporal_model, gti):
 
         return table
 
-    def sample_sources(self, dataset):
+    def sample_sources(self, dataset, psf_update=None):
```

Comment:
It might be simpler if `update_psf` is a bool.
If `True`, pass the `evaluator.update` will take `dataset.psf`, else None